dist/src/main/java/io/camunda/identity/webapp/controllers/IdentityIndexController.java

```diff
@@ -7,26 +7,48 @@
  */
 package io.camunda.identity.webapp.controllers;
 
+import io.camunda.security.configuration.SecurityConfiguration;
+import io.camunda.security.entity.AuthenticationMethod;
 import io.camunda.webapps.controllers.WebappsRequestForwardManager;
 import jakarta.servlet.ServletContext;
 import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
 import java.io.IOException;
-import org.springframework.beans.factory.annotation.Autowired;
+import java.security.SecureRandom;
+import java.util.Base64;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 
 @Controller
 public class IdentityIndexController {
+  private static final String CONTENT_SECURITY_POLICY_HEADER = "Content-Security-Policy";
 
-  @Autowired private ServletContext context;
+  private final ServletContext context;
 
-  @Autowired private WebappsRequestForwardManager webappsRequestForwardManager;
+  private final WebappsRequestForwardManager webappsRequestForwardManager;
+
+  private final SecurityConfiguration securityConfiguration;
+
+  public IdentityIndexController(
+      final ServletContext context,
+      final WebappsRequestForwardManager webappsRequestForwardManager,
+      final SecurityConfiguration securityConfiguration) {
+    this.context = context;
+    this.webappsRequestForwardManager = webappsRequestForwardManager;
+    this.securityConfiguration = securityConfiguration;
+  }
 
   @GetMapping("/identity")
-  public String identity(final Model model) throws IOException {
+  public String identity(final Model model, final HttpServletResponse response) throws IOException {
+    final var envJsNonce = generateNonce();
     model.addAttribute("contextPath", context.getContextPath() + "/identity/");
+    model.addAttribute("clientConfigNonce", envJsNonce);
+    model.addAttribute(
+        "isOidc",
+        securityConfiguration.getAuthentication().getMethod() == AuthenticationMethod.OIDC);
```

Comment:
🙃 Usually, the following is the recommended best practice, but I might be wrong:

```suggestion
        AuthenticationMethod.OIDC.equals(securityConfiguration.getAuthentication().getMethod()));
```